src/deferred/exceptionHook.js

```diff
@@ -0,0 +1,21 @@
+define( [
+	"../core",
+	"../deferred"
+], function( jQuery ) {
+
+// These usually indicate a programmer mistake during development,
+// warn about them ASAP rather than swallowing them by default.
+var rerrorNames = /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;
+
+jQuery.Deferred.exceptionHook = function( error ) {
+
+	// Support: IE9
+	// Console exists when dev tools are open, which can happen at any time
+	var warn = window.console && window.console.warn;
+
+	if ( error && rerrorNames.test( error.name ) ) {
+		warn( "jQuery.Deferred exception: " + error.message );
```

Comment:
Because the Deferred is [resolved through a setTimeout](https://github.com/jquery/jquery/blob/e077ffb083743f4a4b990f586c9d25d787e7b417/src/deferred.js#L185) it loses the stack. We could use some chicanery to save a stack beforehand but that would impact size and performance for the comparatively rare case where an exception occurs. So I pulled out the stack trace entirely since it only refers to things inside jQuery itself.